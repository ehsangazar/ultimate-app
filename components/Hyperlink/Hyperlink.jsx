/** @jsx jsx */
import { css, jsx } from '@emotion/core'
import Link from 'next/link'
import { useTheme } from 'emotion-theming'

const Hyperlink = ({ href, children, ...props }) => {
  const theme = useTheme()
  return (
    <Link href={href}>
      <a
        css={css`
          text-decoration: none;
          cursor: pointer;
          color: ${theme.colors.linkPrimary};
          border-color: ${theme.colors.linkPrimary};
          border-bottom: ${theme.border[1]} solid;
          transition-duration: 0.4s;
          &:hover {
            color: ${theme.colors.linkSecondary};
            border-color: ${theme.colors.linkSecondary};
          }
        `}
        {...props}
      >
        {children}
      </a>
    </Link>
  )
}

export default Hyperlink
