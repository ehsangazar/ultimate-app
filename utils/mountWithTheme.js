import React from 'react'
import { ThemeProvider } from 'emotion-theming'
import renderer from 'react-test-renderer'
import theme from '../configs/theme'

const mountWithTheme = (children) => {
  return renderer.create(
    <ThemeProvider theme={theme}>{children}</ThemeProvider>
  )
}

export default mountWithTheme
