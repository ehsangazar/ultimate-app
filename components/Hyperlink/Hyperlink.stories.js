import React from 'react'
import Hyperlink from './Hyperlink'

export default { title: 'Hyperlink' }

export const simple = () => <Hyperlink href="/">لینک به مطالب</Hyperlink>
