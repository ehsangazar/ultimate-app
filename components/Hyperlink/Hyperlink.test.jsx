import React from 'react'
import serializer from 'jest-emotion'
import mountWithTheme from '../../utils/mountWithTheme'
import Hyperlink from './Hyperlink'

expect.addSnapshotSerializer(serializer)

describe('Hyperlink', () => {
  test('Renders Correctly', () => {
    const component = mountWithTheme(
      <Hyperlink href="/">Sample Hyperlink</Hyperlink>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  })
})
