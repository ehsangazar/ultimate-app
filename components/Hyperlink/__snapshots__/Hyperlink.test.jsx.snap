// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hyperlink Renders Correctly 1`] = `
.emotion-0 {
  -webkit-text-decoration: none;
  text-decoration: none;
  cursor: pointer;
  color: #2c3e50;
  border-color: #2c3e50;
  border-bottom: 2px solid;
  -webkit-transition-duration: 0.4s;
  transition-duration: 0.4s;
}

.emotion-0:hover {
  color: #f1c40f;
  border-color: #f1c40f;
}

<a
  className="emotion-0"
  onClick={[Function]}
  onMouseEnter={[Function]}
>
  Sample Hyperlink
</a>
`;
